src/main/java/io/github/xcube16/iseedragons/FakeThirstStatHandler.java

```diff
@@ -0,0 +1,128 @@
+package io.github.xcube16.iseedragons;
+
+import java.lang.reflect.Method;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+import net.minecraftforge.event.entity.living.LivingEvent.LivingJumpEvent;
+import net.minecraftforge.event.entity.living.LivingHurtEvent;
+import net.minecraftforge.event.entity.player.AttackEntityEvent;
+import net.minecraftforge.event.world.BlockEvent;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+public class FakeThirstStatHandler
+{
+	Object handler;
+	boolean initialized = false;
+	Method onPlayerJump;
+	Method onPlayerHurt;
+	Method onBlockBreak;
+	
+	Method getThirstData;
+	
+	Class thirstHandlerClass;
+	Method addExhaustion;
+	
+	public FakeThirstStatHandler(Object handler)
+	{
+		this.handler = handler;
+		try
+		{
+			onPlayerJump = handler.getClass().getMethod("onPlayerJump", LivingJumpEvent.class);
+			onPlayerHurt = handler.getClass().getMethod("onPlayerHurt", LivingHurtEvent.class);
+			onBlockBreak = handler.getClass().getMethod("onBlockBreak", BlockEvent.BreakEvent.class);
+			
+			getThirstData = Class.forName("toughasnails.api.thirst.ThirstHelper").getMethod("getThirstData", EntityPlayer.class);
+			
+			thirstHandlerClass = Class.forName("toughasnails.thirst.ThirstHandler");
+			addExhaustion = thirstHandlerClass.getMethod("addExhaustion", float.class);
+			
+			//No exceptions, report as successfully initialized
+			initialized = true;
+		}
+		catch(Exception e)
+		{
+			ISeeDragons.logger.error("Failed to initialize FakeThirstStatHandler");
+		}
+	}
+	
+	@SubscribeEvent
+	public void onPlayerJump(LivingJumpEvent event)
+	{
+		if(!initialized)
+			return;
+		try 
+		{
+			onPlayerJump.invoke(handler, event);
+		} 
+		catch (Exception e) 
+		{
+			failMethod(e);
+		}
```

Comment:
If there is a reflection exception, we have created a state in witch ThirstStatHandler will not get any events at all (unless we put the real ThirstStatHandler back on the event bus and get out of the way). In this case we might want to hard stop the game. If it is not a reflection exception, but an exception that came form the real ThirstStatHandler, the expected behaviour is what forge would do when such a thing happens in an event handler (I think it just logs the exception and does not fuse off the handler; could get spammy, but that is normal).